app/services/cold_storage_backends/factory.py

```
@@ -0,0 +1,20 @@
+
+_BACKENDS = {
+    ColdStorageBackendType.LOCAL: LocalColdStorageBackend(),
+    ColdStorageBackendType.S3: S3ColdStorageBackend(),
+    ColdStorageBackendType.GDRIVE: GoogleDriveColdStorageBackend(),
+}
+
+
+def get_backend(location: ColdStorageLocation):
+    backend = _BACKENDS.get(location.backend_type)
+    if backend is None:
+        raise ValueError(f"Unsupported cold storage backend type: {location.backend_type}")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)